lib/rex/ui/text/shell.rb

```diff
@@ -149,8 +149,14 @@ def run(&block)
         # If a block was passed in, pass the line to it.  If it returns true,
         # break out of the shell loop.
         elsif block
+          if self.histfile && line != @last_line
+          	File.open(self.histfile, "a+") { |f| f.puts(line) }
+          	@last_line = line
+          end
+          self.stop_count = 0
```

Comment:
I'm not sure what's going on here. Is this history not able to be managed by Readline?

Reply:
It is being managed by `readline` also but that is for memory not disk. The already present code for managing `msfconsole`'s history also does this. https://github.com/rapid7/metasploit-framework/blob/5f1e5da4a737cba2373c194ff2d474fdb8d8ed7c/lib/rex/ui/text/shell.rb#L160